.github/workflows/osx.yml

```diff
@@ -0,0 +1,42 @@
+name: github-OSX
+
+on: [push, pull_request]
+
+jobs:
+  osxci:
+    name: osx-ci
+    runs-on: [macos-latest]
+
+    strategy:
+      matrix:
+        include:
+          - BACKEND: "SERIAL"
+            BUILD_TYPE: "RelWithDebInfo"
+          - BACKEND: "PTHREAD"
+            BUILD_TYPE: "RelWithDebInfo"
+          - BACKEND: "SERIAL"
+            BUILD_TYPE: "Debug"
+          - BACKEND: "SERIAL"
+            BUILD_TYPE: "Release"
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: configure
+        run: |
+          mkdir build
+          cd build
+          cmake .. \
+            -DKokkos_ENABLE_${{ matrix.BACKEND }}=On \
+            -DCMAKE_CXX_FLAGS="-Werror" \
+            -DCMAKE_CXX_STANDARD=14 \
+            -DKokkos_ENABLE_COMPILER_WARNINGS=ON \
+            -DKokkos_ENABLE_TESTS=On \
+            -DCMAKE_BUILD_TYPE=${{ matrix.CMAKE_BUILD_TYPE }} \
+      - name: build
+        working-directory: build
+        run: |
```

Comment:
No `|` needed if it is one command, also `cmake --build build --parallel 2` should do that same.